File: domain/model/user/user_test.go
package user_test

import (
	"testing"

	"github.com/jupemara/ddd-guys/go/domain/model/user"
)

func TestNewUser_ChangeName(t *testing.T) {
	user, err := user.NewUser("John", "Smith")
	if err != nil {
		t.Errorf("error message: %v", err)
	}
	cases := map[string]struct {
		FirstName        string
		LastName         string
		ExpectedFullName string
		ExpectedError    bool
	}{
		"valid in Japanese": {"太郎", "山田", "太郎 山田", false},
		"empty last name":   {"John", "", "", true},
		"empty first name":  {"", "Smith", "", true},
		"all of empty":      {"", "", "", true},
	}
	for k, c := range cases {
		err := user.ChangeName(c.FirstName, c.LastName)
		if (err != nil) != c.ExpectedError {
			t.Errorf("%s: %v", k, err)
		}
		// 変更された結果を確かめてるだけなので、ここ以下のテストはなくてもいいかもとは思います
		if err == nil {
			fullName := user.Name().FullName()
			if fullName != c.ExpectedFullName {
				t.Errorf("%s: unexpected fullname. actual: %s", k, fullName)
			}
		}
	}
}

File: py/domain/model/user/user_test.py
from domain.model.user.user import User
from domain.model.user.id import Id
import pytest

def test_user():
    cases = {
        "valid user": ["John", "Smith"],
        "valid user in Japanese": ["太郎", "山田"],
    }
    for k in cases:
        case = cases[k]
        user = User(
            Id(k),
            case[0],
            case[1],
        )
        assert user.id().value() == k
        assert user.name().first_name == case[0]
        assert user.name().last_name == case[1]

def test_user_valid():
    user = User(
        Id("id"),
        "John",
        "Smith",
    )
    assert user.id().value() == "id"
    assert user.name().first_name == "John"
    assert user.name().last_name == "Smith"

def test_user_error():
    cases = {
        "empty fist_name": ["", "non-empty"],
        "empty last_name": ["non-empty", ""],
    }
    for k in cases:
        with pytest.rases(AssertionError):
            case = cases[k]
            User(
                Id(k),
                case[0],
                case[1],
            )
